### ISSUE TITLE
Cleanup indexer exports

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#98]
- We use several crates in the indexer project
  - fuel-indexer-schema, fuel-indexer-derive, fuel-executor, fuel-indexer, fuel-api-server
- We export various components out of these crates, but in some cases, you can't use an exported component unless you include one of the component's dependencies
  - e.g., fuel-indexer is required to use fuel-indexer-derive's `graphql_schema!` macro
    - If the use of `graphql_schema!` requires fuel-indexer, it should be exported out of fuel-indexer
- Imports that require other fuel-indexer components, should be exported out of the fuel-indexer
  - `fuel-indexer` should re-export everything, since all components constitute a Fuel Indexer
    - Obviously you would still be able to import without going through the fuel-indexer (e.g., `use fuel_indexer::*`)

Additional context

An ideal import might look like:

```rust
// Gives me everything I need from a single source, while still allowing me to import
// via separate crates if I so choose to do so
use fuel_indexer::{
    executor::{GraphQlApi, IndexerConfig, IndexerResult, IndexerService}, 
    derive::{graphql_schema, abigen},
    types::Address,
};

// Also allows me to do

use fuel_indexer_derive::{graphql_schema, abigen};
use fuel_executor::{GraphQlApi, IndexerConfig, IndexerResult, IndexerService};
use fuel_indexer_schema::types::Address;
```
- This example shows that the `fuel_indexer` depends on the `executor`, but the `exectuor` can also be used in a standalone `fuel_exector` crate
- There is no case where you would want to use `fuel_indexer` without `fuel_executor`
- Again, this is just an example

**Comment by ra0x3** [comment]
@tjsharp1 Wanna get your opinion on this.

TLDR; How do you feel about being able to import everything through `fuel_indexer` ?
